merk/src/proofs/query/mod.rs

```diff
@@ -70,6 +70,20 @@ pub struct SubqueryBranch {
     pub subquery: Option<Box<Query>>,
 }
 
+impl SubqueryBranch {
+    /// Returns the depth of the subquery branch
+    /// This depth is how many GroveDB layers down we could query at maximum
+    #[inline]
+    pub fn max_depth(&self) -> u32 {
+        let subquery_path_depth = self
+            .subquery_path
+            .as_ref()
+            .map_or(0, |path| path.len() as u32);
+        let subquery_depth = self.subquery.as_ref().map_or(0, |query| query.max_depth());
```

Comment:
if this check will be called for externally received (malicious) queries it can cause a grovedb panic with recursion limit

Reply:
Should now be fixed